crates/core/src/transport/crypto.rs

```diff
@@ -89,7 +90,9 @@ impl TransportSecretKey {
         self.0.decrypt(padding, data)
     }
 
-    pub fn to_bytes(&self) -> Result<Vec<u8>, pkcs1::Error> {
+    pub fn to_pkcs1_pem(&self) -> Result<Vec<u8>, pkcs1::Error> {
```

Comment:
remove all references to pkcs1 altogether and let's assert (via unit test) that the transport works correctly using PKCS8 and it's corresponding public key.